lua/nvim-tree/actions/git/cli.lua

```diff
@@ -0,0 +1,89 @@
+local notify = require "nvim-tree.notify"
+local log = require "nvim-tree.log"
+-- vim.loop got renamed in 0.10.0
+-- ref: https://github.com/neovim/neovim/pull/22846
+local uv = vim.loop or vim.uv
+
+local M = {}
+--- See h uv.spawn-options for full list
+--- @class uv.SpawnOptions
+--- @field args string[] Args to be passed to git command e.g. -C
+--- @field env table|nil
+--- @field cwd table|nil
+--- @field detached boolean|nil Detach process from parent
+--- @field timeout number|nil
+
+--- Generic git job command powered by libuv
+--- Unlie git/runner.lua, it provides generic interface to the git comand output
+--- @param uvSpawnOptions uv.SpawnOptions uv.spaw like options
+--- @param onErrOrData fun(err: nil|string, data: string) error handler callback
+--- @param onExit? fun(code: number, signal: number) error handler callback
+function M.git(uvSpawnOptions, onErrOrData, onExit)
+
+  if type(uvSpawnOptions.args) ~= "table" then
+    error("uvSpawnOptions.args is required", 2)
+  end
+
+  local onErrorOrData_type = type(onErrOrData)
+  if onErrorOrData_type ~= "function" then
+    error("onErrorOrData callback is required, got " .. onErrorOrData_type, 2)
+  end
+
+  local stdin = uv.new_pipe()
+  local stdout = uv.new_pipe()
+  local stderr = uv.new_pipe()
```

Comment:
This functionality appears to duplicate `git/runner.lua`.

Let's not repeat ourselves and share that code.

Reply:
Please merge:
https://github.com/nvim-tree/nvim-tree.lua/pull/2551